Packages/com.unity.inputsystem/InputSystem/API/Input.cs

```diff
@@ -0,0 +1,1862 @@
+        {
+            for (var i = 0; i < maxGamepadSlots; i++)
+            {
+                s_Gamepads[i] = null;
+                s_GamepadsConnectedFrames[i] = -1;
+                s_GamepadsDisconnectedFrames[i] = -1;
+            }
+
+            for (var i = 0; i < (int)InputSlot.Joystick_Max; i++)
+            {
+                s_Joysticks[i] = null;
+            }
+
+            InputSystem.onDeviceChange -= OnDeviceChange;
+            InputSystem.onDeviceChange += OnDeviceChange;
+
+            InitializeGamepadConfigs();
+
+            // add any existing gamepads
+            foreach (var gamepad in Gamepad.all)
+            {
+                AddGamepadToFirstFreeSlot(gamepad);
+            }
+
+            foreach (var joystick in Joystick.all)
+            {
+                AddJoystickToFirstFreeSlot(joystick);
+            }
+
+#if UNITY_EDITOR
+            // To support DidGamepadConnectThisFrame and DidGamepadDisconnectThisFrame, we make use of Time.frameCount
+            // to record what frame a device was connected or disconnected on. A problem with this is that in the editor,
+            // device connected events can come from an editor update that happens before Time.frameCount has been
+            // updated for the frame, and from the managed side, we have no way to tell that we're inside that update.
+            // An additional complication is that device connected events happen asynchronously on the native side, so
+            // sometimes the event will come through the editor update and sometimes through the normal player loop.
+            // So what we do is install two additional PlayerLoop systems, one to track when the Time.frameCount value has
+            // been updated and one to reset at the end of the frame. If we end up adding a device before the frame
+            // count has been updated, we manually add 1 to the frame count that we record.
+            var playerLoop = UnityEngine.LowLevel.PlayerLoop.GetCurrentPlayerLoop();
+            playerLoop.InsertSystemAsSubSystemOf<InputSystemPlayerLoopHighLevelTimeUpdate, TimeUpdate>(
+                () => s_TimeHasUpdatedThisFrame = true);
+            playerLoop.InsertSystemAsSubSystemOf<InputSystemPlayerLoopHighLevelEndFrame, PostLateUpdate>(
+                () => s_TimeHasUpdatedThisFrame = false);
+            UnityEngine.LowLevel.PlayerLoop.SetPlayerLoop(playerLoop);
+#endif
+            s_PointerPosition = Vector2.zero;
+            s_ScrollDelta = Vector2.zero;
+
+            s_PointerAction = new InputAction("Pointer", InputActionType.PassThrough, expectedControlType: "Vector2");
+            s_PointerAction.AddBinding("<Pointer>/position");
+            s_PointerAction.performed += OnPointerMoved;
+            s_PointerAction.Enable();
+
+            s_ScrollAction = new InputAction("Scroll", InputActionType.PassThrough, expectedControlType: "Vector2");
```

Comment:
I'm wondering if this forces mouse processing even if we never use it.
Something to check on performance side later.